### PR TITLE
Train test split

### PR DESCRIPTION
**Pull request by owain-Biddulph**

### CODE REVIEW

File: split.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# Train/test split 80/20, stratify on highest ranked emotion
+train_x, test_x, train_df_y, test_df_y = train_test_split(
+    all_spectro, labels, train_size=0.8,  stratify=labels.idxmax(axis=1))
+
+# Dump split data
+utils.dump_elements(train_x, config.TRAIN_DATA_PATH)
+train_df_y.to_pickle(config.TRAIN_LABELS_PATH)
+utils.dump_elements(test_x, config.TEST_DATA_PATH)
+test_df_y.to_pickle(config.TEST_LABELS_PATH)
```
Comment (quentinverlhac) [inline split.py]: I think all of this should be written in one or more function(s) that you can call in `if __name__ == '__main__':`. This would keep the same logic running when you execute the current file, but it would also allow us to call this logic from other files.

File: split.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+from pathlib import Path
+
+import config
+import utils
+
+# Create split_data directory if it doesn’t exist
+Path(config.SPLIT_DATA_DIRECTORY_PATH).mkdir(parents=True, exist_ok=True)
+
+# Load processed data
+with open(config.EMOTIFY_SPECTROGRAM_DUMP_PATH, "rb") as f:
```
Comment (quentinverlhac) [inline split.py]: For information, I have added in my PR a function that does the loading (you just have to provide the path). I will change these lines in my PR if you merge yours before. Otherwise could you use this function here ?

File: split.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    all_spectro = pkl.load(f)
+
+with open(config.EMOTIFY_LABELS_DUMP_PATH, "rb") as f:
+    labels = pkl.load(f)
+
+# Transform labels to pandas DataFrame (mainly to use idxmax)
+labels = pd.DataFrame(labels)
+labels.columns = config.EMOTIFY_EMOTIONS_ORDERED_LIST
```
Comment (quentinverlhac) [inline split.py]: It would be nice to write this in a function so we can call it every time we need a `labels` dataframe. Can be done in another PR.

File: split.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+# Train/test split 80/20, stratify on highest ranked emotion
+train_x, test_x, train_df_y, test_df_y = train_test_split(
+    all_spectro, labels, train_size=0.8,  stratify=labels.idxmax(axis=1))
```
Comment (quentinverlhac) [inline split.py]: Train size could be in `config.py`
